v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/utils/ElasticsearchIO.java

```diff
@@ -1452,7 +1452,8 @@ public void processElement(ProcessContext context) throws Exception {
                     "{ \"update\" : %s }%n{ \"doc\" : %s, \"doc_as_upsert\" : true }%n",
                     documentMetadata, document));
           } else {
-            batch.add(String.format("{ \"create\" : %s }%n%s%n", documentMetadata, document));
+            // switch from create to index to allow upsert (beneficial in case of duplication due to timeout/retry)
+            batch.add(String.format("{ \"index\" : %s }%n%s%n", documentMetadata, document));
```

Comment:
Makes sense. This matches https://github.com/apache/beam/blob/master/sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java#L1576

Reply:
is a change necessary or can I resolve this?